scoap3/modules/records/cli.py

```diff
@@ -65,4 +88,138 @@ def loadrecords(source):
             queue=queue,
         )
 
-        click.echo('Parsed record {}.'.format(i))
+        info('Parsed record {}.'.format(i))
+
+
+@click.group()
+def fixdb():
+    """Database fix commands."""
+
+
+@fixdb.command()
+@with_appcontext
+def unescaperecords():
+    """HTML unescape abstract and title for all records."""
+
+    parser = HTMLParser()
+
+    def proc(record, parser):
+        if record.json is None:
+            rerror('record.json is None', record)
+            return
+
+        unescape_abstract(record, parser)
+        unescape_titles(record, parser)
+
+    process_all_records(proc, 50, parser)
+
+    info('all done!')
+
+
+def unescape_abstract(record, parser):
+    if 'abstracts' not in record.json or len(record.json['abstracts']) == 0:
+        rerror('Record has no abstracts.', record)
+        return
+
+    if len(record.json['abstracts']) > 1:
+        rerror('Record has more then one abstracts (%d). Skipping.' % len(record.json['abstracts']), record)
+        return
+
+    original = record.json['abstracts'][0]['value']
+    unescaped = parser.unescape(original)
+    if unescaped != original:
+        rinfo('Abstract changed.', record)
+        record.json['abstracts'][0]['value'] = unescaped
+        flag_modified(record, 'json')
+
+
+def unescape_titles(record, parser):
+    if 'titles' not in record.json or len(record.json['titles']) == 0:
+        rerror('Record has no titles.', record)
+        return
+
+    original = record.json['titles']
+    unescaped = []
+
+    for title in original:
+        if 'title' not in title:
+            rerror('title key not in title', record)
+
+        title['title'] = parser.unescape(title['title'])
+        unescaped.append(title)
+
+    if unescaped != original:
+        rinfo('Authors changed.', record)
+        record.json['titles'] = unescaped
+        flag_modified(record, 'json')
+
+
+def utf8rec(data):
+    if isinstance(data, basestring):
+        try:
+            return ''.join(chr(ord(c)) for c in data).decode('utf8')
+        except:
+            return data
+
+    if isinstance(data, tuple) or isinstance(data, list):
+        return [utf8rec(element) for element in data]
+
+    if isinstance(data, dict):
+        return {k: utf8rec(v) for k, v in data.items()}
+
+    if isinstance(data, numbers.Number) or data is None:
+        return data
+
+    error('Couldn\'t determine the data type of %s. Returning the same.' % data)
+    return data
+
+
+@fixdb.command()
+@with_appcontext
+def utf8():
+    """Unescape records and store data as unicode."""
+
+    def proc(record):
+        if record.json is None:
+            rerror('record.json is None', record)
+            return
+        record.json = utf8rec(record.json)
+        flag_modified(record, 'json')
+
+    process_all_records(proc)
+    info('all done!')
+
+
+def process_all_records(function, chuck_size=50, *args):
```

Comment:
Allow to pass a list of record ids (control_number), if empty list then run on all, or something like this